### PR TITLE
Add vector variable to FIMS

### PR DESCRIPTION
**Pull request by ChristineStawitz-NOAA**
# What is the feature?

* This addresses issue #388 and #584 by allowing parameters/derived quantities to be vectors instead of scalars

# How have you implemented the solution?

* Add ParameterVector to rcpp_interface_base.hpp
* export ParameterVector to R in rcpp_inerface.hpp
* add operators.R file to perform operators on R objects
* add test-unit-rcpp-interface-variable-vector.R to test the ParameterVector

# Does the PR impact any other area of the project?

This will be used throughout FIMS to set parameter values

# How to test this change

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

File: inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp [hunk 1]
```diff
@@ -35,26 +37,160 @@ class Parameter {
...
+
+    void set_all_random(bool random){
+        for(size_t i = 0; i < this->storage_m.size(); i++){
+            Parameter p = Rcpp::as<Parameter>(this->storage_m[i]);
+            p.random_m = random;
+            this->storage_m[i] = Rcpp::wrap(p);
+        }
+    }
+    
+    void fill(double value){
```
Comment (ChristineStawitz-NOAA) [inline inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp:176]: Documentation needed

File: inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp [hunk 2]
```diff
@@ -35,26 +37,160 @@ class Parameter {
...
+    
+    void set_all_estimable(bool estimable){
+        for(size_t i = 0; i < this->storage_m.size(); i++){
+            Parameter p = Rcpp::as<Parameter>(this->storage_m[i]);
+            p.estimated_m = estimable;
+            this->storage_m[i] = Rcpp::wrap(p);
+        }
+    }
+
+    void set_all_random(bool random){
```
Comment (ChristineStawitz-NOAA) [inline inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp:168]: Needs documentation

File: inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp [hunk 3]
```diff
@@ -35,26 +37,160 @@ class Parameter {
...
+            this->storage_m[i] = Rcpp::wrap(p);
+        }
+    }
+    
+};
+uint32_t ParameterVector::id_g = 0;
+
+
+
+ParameterVector CreateVector(size_t size){
```
Comment (ChristineStawitz-NOAA) [inline inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp:189]: Needs documentation

File: tests/testthat/test-unit-rcpp-interface-variable-vector.R [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+p[1]$value
+p[1]$value <- 1
+p[1]$value
+k <- p[1]$id
+for(i in 2:length(p)){
+   expect_equal(p[i]$id, k + 1)
+   k <- p[i]$id
+}
+
+#Test that resize works
```
Comment (ChristineStawitz-NOAA) [inline tests/testthat/test-unit-rcpp-interface-variable-vector.R:70]: This is as far as we got on 5/28

**Comment by Andrea-Havron-NOAA** [comment]
Replaced with PR #604

**Pull request closed without merging**